src/Microsoft.AspNetCore.Server.HttpSys/RequestProcessing/RequestStream.cs

```diff
@@ -177,6 +166,10 @@ public override unsafe int Read([In, Out] byte[] buffer, int offset, int size)
                 }
                 UpdateAfterRead(statusCode, dataRead);
             }
+            if (TryCheckSizeLimit((int)dataRead, out var ex))
```

Comment:
`// Note SwitchToOpaqueMode sets HasStarted and clears _maxSize, so these limits don't apply.`